### DISCUSSION TITLE
Band structure plot for Hybrid

### DISCUSSION THREAD

**Discussion by jblls** [SMTG-UCL/sumo#165] [Q&A] [answered]
Hi All,

I would like to ask a question about plotting band structures for hybrid calculations
(apologies if it is a trivial question, I am not a VASP expert and I just started using sumo).

The problem I am having is that the band structure I am generating for HSE looks discontinuous,
the bands are continuous along the generated zero-weight kpath, but there seems to be "extra" paths in between
(see attached file).

I know that sumo should already eliminate the non-zero weight kpoints.
I think the problem is that the k-distance generated by sumo seems to jump or duplicate,
for example in the file band.dat:

#k-distance eigenvalue[eV]
0.00000000 -0.34800000
0.02836263 -0.37400000
0.05672525 -0.44330000
0.08508788 -0.53730000
...
0.83403927 -0.38930000
0.87053459 -0.34800000  <--
0.87053459 -0.34800000  <--
0.91438906 -0.33650000
...
1.13366143 -0.02540000
1.17751590 0.00000000.    <--
1.53998983 -0.81960000. <--
1.57339558 -0.82280000

I was wondering if this could be due to the fact that I am generating the KPOINTS file with vaspkit
which in its new version uses an uneven point density along different directions,
e.g.
non 0-weight            |     0-weight  with different number of points per k-path 
0.030   3   3   3   14  0.030   78    7   10    8    8   15   11   13   13   # Parameters to Generate KPOINTS (Do NOT Edit This Line)
https://vaspkit.com/tutorials.html#pre-process-band-structure-hybrid-functional

If this is indeed the problem,
Is there an easy fix?

Thank you,

![band_1](https://user-images.githubusercontent.com/22128734/167850208-15e61ee0-b947-470a-936b-ac32d06cea7c.jpg)

**Comment by ajjackson** [comment, marked as answer]
Hi jblls,

The k-points choice does not look incorrect, but Sumo does not now how to plot data from vaspkit; _bandplot_ was primarily designed for use with _kgen_.

Where a k-point appears twice, this usually indicates a (high-symmetry, labelled) turning point in the band path.
Where there is a large gap, this would typically be plotted as a discontinuous break in the band plot without lines between them. These are necessary to efficiently cover the high-symmetry lines without much doubling-back or redundancy.

Sumo relies on high-symmetry labels being added as comments in order to label the x-axis; because they are not in the  KPOINTS file from vaspkit, it plots the data "naively", with x-axis spacing appropriate to the distance between those points. If you look at a typical file from _kgen_ and happen to know what the symmetry labels should be, you might be able to adapt the KPOINTS file for use with Sumo. (It would however be simpler to use _kgen_ in the first place! It does have features to support hybrid-xc calculations.)

**Reply by jblls** [reply]
Hi Adam,

thank you for replying so quickly,

I understand the issue now,
it is just that all my band paths (normal DFT and HSE) were generated with vaspkit, 
and only for HSE (in which the path is given explicitly, and not in Line-Mode)
I got this problem.

As a quick fix, I actually used: 
sumo-kgen --seekpath (which is the same algorithm used by vaspkit),
and used the sumo generated KPOINTS file just plotting, 
and that fixed the issue.

Thank you for your help.
Stefano

**Reply by ajjackson** [reply]
If doing that, I would check _very_ carefully that the kpoints are truly the same, as there could be some different conventions around path interpolation distances, duplication at high-symmetry points etc.

I don't recommend ever using Line Mode for a band structure with more than one direction, because it gives inconsistent sampling density along sections of different length.

**Reply by jblls** [reply]
Hi Adam,

thank you for your help,
and your suggestions.

I will repeat the calculations using sumo-kgen for consistency,

but just for the records, manually adding the Labels to the Vaspkit generated KPOINTS file
(without having sumo regenerating the file as I suggested before)
seems to be enough for sumo to recognise the disjoint path, 
I checked the kpath and as far as I can tell the numbers match.

Sorry for wasting your time,
and thanks for the amazing tool,

Stefano

**Comment by ajjackson** [comment]
Glad to hear that you were able to get a decent result by modifying the KPOINTS from Vaspkit. Indeed, this should give sumo-bandplot all the information it needs.

Happy calculating!
